src/Http/CurlDispatcher.php

```diff
@@ -187,6 +197,13 @@ private function writeBody($curl, $string): int
             $this->body = fopen('php://temp', 'w+');
         }
 
+        if ($this->isBinary) {
+            fclose($this->body);
+            $this->body = null;
+
+            return -1;
+        }
+
         return fwrite($this->body, $string);
```

Comment:
If we detect the response is binary we don't have to open a temporary stream in the first place.Something like this would be better:

```php
if ($this->isBinary) {
    return -1;
}

if (!$this->body) {
    $this->body = fopen('php://temp', 'w+');
}

return fwrite($this->body, $string);
```

Reply:
You are absolutely right, must have overlooked this. Thank you!